core/templates/pages/learner-group-pages/edit-group/edit-learner-group-page-auth.guard.spec.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for EditLearnerGroupPageAuthGuard
+ */
+import { Location } from '@angular/common';
+import { TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { ActivatedRouteSnapshot, RouterStateSnapshot, Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { AppConstants } from 'app.constants';
+import { of } from 'rxjs';
+import { EditLearnerGroupPageAuthGuard } from './edit-learner-group-page-auth.guard';
+import { AccessValidationBackendApiService } from 'pages/oppia-root/routing/access-validation-backend-api.service';
+import { ContextService } from 'services/context.service';
+
+
+class MockAccessValidationBackendApiService {
+  validateAccessToLearnerGroupEditorPage(learnerGroupId: string) {
+    return of().toPromise();
```

Comment:
Any specific reason for creating a promise via `of` and not directly `new Promise(...`?

Reply:
not really both do the same thing, removed it.